packages/scanner/src/cli/scan/watchScan.ts

```diff
@@ -32,10 +33,20 @@ export class Watcher {
       .on('add', this.reloadConfig.bind(this))
       .on('change', this.reloadConfig.bind(this));
 
-    this.appmapWatcher = chokidar.watch(`${this.options.appmapDir}/**/mtime`, {
+    // Chokidar struggles with relative paths. Make sure the watch pattern is absolute.
+    const watchPattern = path.resolve(this.options.appmapDir, '**', 'mtime');
+    console.log(watchPattern);
+    this.appmapWatcher = chokidar.watch(watchPattern, {
       ignoreInitial: true,
     });
-    this.appmapWatcher.on('add', this.scan.bind(this)).on('change', this.scan.bind(this));
+    this.appmapWatcher
+      .on('add', (filePath) => this.scan(filePath))
+      .on('change', (filePath) => this.scan(filePath))
+      .on('unlink', (path: string) => {
+        // If you add and delete an mtime file, Chokidar will **sometimes** seemingly remove future
+        // watchers at this path. Re-adding it here resolves the issue.
+        this.appmapWatcher?.add(path);
+      });
```

Comment:
To be totally honest, I'm not 100% sure why this is needed. In some cases, using an absolute path for the watch pattern resolves the missing fs events and other times this is necessary.